ofono/src/gprs.c

```diff
@@ -820,6 +822,185 @@ static void pri_update_mms_context_settings(struct pri_context *ctx)
 	pri_limit_mtu(settings->interface, MAX_MMS_MTU);
 }
 
+static gboolean pri_str_changed(const char *val, const char *newval)
+{
+	return newval ? (strcmp(val, newval) != 0) : !val[0];
```

Comment:
Please consider using e.g.
https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strcmp0
which seems to handle NULL pointers gracefully